src/commands/init.ts

```diff
@@ -192,29 +193,39 @@ export default class Init extends Command {
     }
     packageJson.scripts = { ...packageJson.scripts, ...scripts };
 
-    await this.addDependencies(packageJsonData.dependencies, false, packageJson, flags.projectDir);
-    await this.addDependencies(packageJsonData.devDependencies, true, packageJson, flags.projectDir);
+    packageJson.dependencies = await this.addDependencies(packageJsonData.dependencies, packageJson.dependencies, flags.projectDir);
+    packageJson.devDependencies = await this.addDependencies(packageJsonData.devDependencies, packageJson.devDependencies, flags.projectDir);
 
     fs.writeFileSync(path.resolve(flags.projectDir, 'package.json'), JSON.stringify(packageJson, null, 2));
   }
 
-  private async addDependencies(dependencies: string[], isDev: boolean, packageJson: any, projectDir: string) {
-    packageJson.devDependencies = packageJson.devDependencies ? packageJson.devDependencies : {};
-    packageJson.dependencies = packageJson.dependencies ? packageJson.dependencies : {};
+  private async addDependencies(
+    dependencies: string[],
+    toBeAdded: { [key: string]: string },
+    projectDir: string
+  ): Promise<{ [key: string]: string }> {
+    const copy: { [key: string]: string } = {};
+    if (toBeAdded) {
+      Object.entries(toBeAdded).forEach(([key, value]) => (copy[key] = value));
+    }
     const promises = dependencies.map(async dependency => {
-      try {
-        const latestVersion = (await execa('npm', ['view', dependency, 'version'], { cwd: projectDir })).stdout;
-        if (isDev) {
-          packageJson.devDependencies[dependency] = `^${latestVersion}`;
-        } else {
-          packageJson.dependencies[dependency] = `^${latestVersion}`;
-        }
-      } catch (err) {
-        this.error(`Error in finding version for dependency ${dependency}`);
+      const latestVersion = await this.getVersionOfDependency(dependency, projectDir);
+      if (latestVersion) {
+        copy[dependency] = `^${latestVersion}`;
       }
       return Promise.resolve();
     });
-    return Promise.all(promises);
+    await Promise.all(promises);
```

Comment:
This could be done already in line 211: 
``` ts 
await Promise.all(dependencies.map())
```
Also I am pretty sure that `.map` does not support async functions correctly. Therefore you should not `async await` in the callback (instead just 
``` ts
dep => this.getVersionOfDependency(dep);
``` 
should be enough).